### PR TITLE
Set session_key state, make password field private

### PR DESCRIPTION
**Pull request by ArchKevin22**

### CODE REVIEW

File: client/js/HomeScreen.js [hunk 1]
```diff
@@ -15,15 +15,14 @@ export default class HomeScreen extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            user_id: '',
+            userID: '',
```
Comment (bliu23) [inline client/js/HomeScreen.js]: userId (camelcase)

File: client/js/LoginScreen.js [hunk 2]
```diff
@@ -8,20 +8,26 @@ export default class LoginScreen extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            user_id: '',
+            userID: '',
```
Comment (bliu23) [inline client/js/LoginScreen.js]: userId

File: client/js/LoginScreen.js [hunk 3]
```diff
@@ -30,9 +36,10 @@ export default class LoginScreen extends React.Component {
             <Text style={styles.title}>Conveni-friends</Text>
             <TextInput 
                 placeholder="User ID"
-                onChangeText={(text) => this.setState({user_id: text})}
+                onChangeText={(text) => this.setState({userID: text})}
```
Comment (bliu23) [inline client/js/LoginScreen.js]: userId

File: client/js/SignupScreen.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { Alert, StyleSheet, Text, View, TextInput, Button } from 'react-native';
+
+export default class SignupScreen extends React.Component {
+    static navigationOptions = {
+        //headerLeft: null,
+    }
+    constructor(props) {
+        super(props);
+        this.state = {
+            userID: '',
```
Comment (bliu23) [inline client/js/SignupScreen.js]: etc etc

File: client/js/SignupScreen.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            password2: '',
+            sessionKey: ''
+        };
+        this._makeAccount = this._makeAccount.bind(this)
+    }
+    _makeAccount() {
+        //alphanumeric characters
+        const re = /[0-9a-zA-Z]+/g;
+        //TODO: check w/backend that the user id does not already exist
+        if (this.state.userId && re.test(this.state.userId)) {
```
Comment (bliu23) [inline client/js/SignupScreen.js]: we should move this out to a separate function

if(isValidInput(this.state.userId) {
    this.props.navigation.navigate('Login')
} 

function isValidInput(userId) {
    // check for validation and return if good or not
}

File: client/js/NearbyRequests.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { ListView, Alert, StyleSheet, Text, View, TextInput, Button } from 'react-native';
+
+export default class NearbyRequests extends React.Component {
+    static navigationOptions = {
+        //headerLeft: null,
+    }
+    constructor(props) {
+        super(props);
+        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
```
Comment (bliu23) [inline client/js/NearbyRequests.js]: nit: descriptive variable names 😢

File: client/js/SignupScreen.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            userID: '',
+            password: '',
+            password2: '',
+            sessionKey: ''
+        };
+        this._makeAccount = this._makeAccount.bind(this)
+    }
+    _makeAccount() {
+        //alphanumeric characters
+        const re = /[0-9a-zA-Z]+/g;
```
Comment (bliu23) [inline client/js/SignupScreen.js]: nit: var name :|

**Review by bliu23 — CHANGES_REQUESTED** [review]

File: client/js/HomeScreen.js (hunk 1 above)
Comment (ArchKevin22) [reply client/js/HomeScreen.js]: Done

File: client/js/LoginScreen.js (hunk 3 above)
Comment (ArchKevin22) [reply client/js/LoginScreen.js]: Done

**Review by bliu23 — APPROVED** [review]

**Pull request merged**